test/tests/interfaces/pbs_partition.py

```diff
@@ -0,0 +1,297 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2017 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.interfaces import *
+
+
+class TestPartition(TestInterfaces):
+    """
+    Test suite to test partition attr
+    """
+    host_name = socket.gethostname()
+
+    def partition_attr(self, mgr_cmd="MGR_CMD_SET",
+                       obj_name="QUEUE", q_type="execution",
+                       q_name="Q1", enable="True",
+                       start="True", n_name=host_name,
+                       partition="P1", user=ROOT_USER):
+        """
+        Common function to set partition attribute/s to node/queue object
+        :param mgr_cmd: qmgr "MGR_CMD_SET/MGR_CMD_UNSET/MGR_CMD_CREATE" cmd,
+        Defaults to MGR_CMD_SET
+        :type mgr_cmd: str
+        :param obj_name: PBS object vnode/queue. Defaults to queue
+        :type obj_name: str
+        :param q_type: "queue_type" attribute of queue object ,
+                    Defaults to "execution"
+        :type q_type:  str
+        :param q_name: name of the queue, Defaults to "Q1"
+        :type q_name: str
+        :param enable: "enabled" attribute of queue object, Defaults to "True"
+        :type enable: boolean
+        :param start: "started" attribute of queue object, Defaults to "True"
+        :type start: boolean
+        :param n_name: name of the vnode, Defaults to "shortname of server"
+        :type n_name: str
```

Comment:
I think you can merge q_name and n_name into just name.  You either act upon a queue or a node, but never both.

Reply:
Done